server/src/main/java/org/apache/iotdb/db/metadata/lastCache/LastCacheManager.java

```diff
@@ -85,11 +87,15 @@ public static void updateLastCache(
     checkIsTemplateLastCacheAndSetIfAbsent(node);
 
     ILastCacheContainer lastCacheContainer = node.getLastCacheContainer();
+    if (lastCacheContainer.isEmptyContainer()) {
+      lastCacheContainer = new LastCacheContainer();
+      node.setLastCacheContainer(lastCacheContainer);
+    }
```

Comment:
This code block should be synchronized on the given node. Otherwise, multi threads will do the same operation, and this may lead to some of them losing execution results.